app/src/components/Modals/activity-modal/activity-modal-body.tsx

```diff
@@ -278,6 +278,7 @@ const ActivityModalBody = ({
               <FormControl
                 isInvalid={!!resolve(prefix + "activityDataAmount", errors)}
               >
+                {/*TODO dynamically render required fields*/}
```

Comment:
This is a big and quite important task. Please create a Jira ticket for the next sprint with highest priority so we can finish this properly.

Reply:
Done!